Sources/TuistKit/ProjectEditor/ProjectEditor.swift

```diff
@@ -120,4 +141,38 @@ final class ProjectEditor: ProjectEditing {
         try writer.write(workspace: descriptor)
         return descriptor.xcworkspacePath
     }
+
+    /// - Returns: A list of manifest name and path for plugin manifests that are editable and should be
+    /// loaded as a target in the generated project.
+    private func locateEditablePluginManifests(at path: AbsolutePath, plugins: Plugins) -> [(String, AbsolutePath)] {
+        let editingPluginManifests = manifestFilesLocator.locatePluginManifests(at: path)
+        let loadedHelpers = plugins.projectDescriptionHelpers
+
+        // If a loaded plugin is also a locally editable plugin, we should take the name of the loaded plugin (defined in Plugin.swift manifest)
+        // Otherwise, use the name of the parent directory as the name of the plugin.
+        return editingPluginManifests.map { editableManifest in
+            if let loadedEditableManifest = loadedHelpers.first(where: { $0.path.parentDirectory == editableManifest.parentDirectory }) {
+                return (loadedEditableManifest.name, editableManifest)
+            } else {
+                return (editableManifest.parentDirectory.basename, editableManifest)
+            }
+        }
+    }
+
+    private func buildPluginModules(
+        in path: AbsolutePath,
+        projectDescriptionPath: AbsolutePath,
+        editablePluginManifestPaths: [AbsolutePath],
+        plugins: Plugins
+    ) throws -> [ProjectDescriptionHelpersModule] {
+        let loadedPluginHelpers = plugins.projectDescriptionHelpers.filter { loadedHelper in
+            !editablePluginManifestPaths.contains(where: { $0.parentDirectory == loadedHelper.path.parentDirectory })
```

Comment:
Had to modify the way `PluginService` returns the loaded plugins since we had previously lost this information (local vs. remote) during translation from description to graph model, but as you mentioned this is worth it so we can support editing all local plugins regardless of their location (as long as they can be read from Config.swift)